Changed e-file module and removed the multiple calls to pfl

bananas-repos opened the pull request:
This PR does now reflect the current develop changes made with https://github.com/portagefilelist/client/commit/037d214c48d8446b47d8e707fbb30be04800aea4 There I've changed the modules to be more module-like and not a patched up version of something summed up over the years.

The changes to **command-not-found** are minimal. Renamed to the new imports and removed the loop around calling pfl. Pfl does a good job already by searching for a command and will return multiple results. There is no need to do multiple calls to pfl. Does make it faster, too.

bananas-repos committed:
now matching the current develop version of e-file
removed loop since e-file does already a good search. There is no need to try with different paths and multiple requests

Nowa-Ammerlaan reviewed (commented):
Thanks for the PR!

Nowa-Ammerlaan commented on command-not-found:
```diff
@@ -94,23 +95,18 @@ def main():
-            print("")
-        else:
-            print("No candidates found\n\n")
+        try:
+            options = {
+                'file': cmd
+            }
+            ret, out = pfl.e_file.run(options)
+            if ret == 0:
+                found = True
+                sys.stdout.write(out)
+                print("")
+        except SystemExit:
+            # If something is not working just quit
+            print("No candidates found\n\n")    
```
But this is going to match every package that installs a file named `cmd` no matter where it is installed. For this use case we are only interested in packages that install `cmd` into `$PATH`. That is why the original code searches specifically for whole paths not just the filename.

This would also introduce a difference in behaviour between the direct importing method and the compatibility fallback method where we just call `/usr/bin/e-file` directly.

bananas-repos committed:
revert the remove of the loop. Based on feedback from owner

bananas-repos commented:
I've updated the PR and added the loop again. Currently there is now way to query pfl with multiple params. Maybe in the future. This would reduce the calls you have to make to one.

AndrewAmmerlaan commented:
> I've updated the PR and added the loop again. Currently there is now way to query pfl with multiple params. Maybe in the future. This would reduce the calls you have to make to one.

Thanks! :+1: 

Could you give me a ping when you release the next version of `pfl`? I'll then release a new version of `command-not-found` shortly after.

bananas-repos commented:
https://github.com/portagefilelist/client/releases/tag/3.5 - release today.
Version bump made: https://bugs.gentoo.org/926417
